src/profile-logic/profile-data.js

```diff
@@ -879,38 +880,41 @@ export function filterThreadToSearchString(
 }
 
 /**
- * This function takes both a SamplesTable and can be used on CounterSamplesTable.
+ * This function takes a SamplesTable and can also be used on CounterSamplesTable.
+ * It takes a time-based range as parameter and returns a sample-based range.
+ * Just like rangeStart is inclusive and rangeEnd is exclusive, sampleStart is
+ * inclusive and sampleEnd is exclusive.
  */
-function _getSampleIndexRangeForSelection(
+export function getSampleIndexRangeForSelection(
   samples: SamplesTable | CounterSamplesTable,
   rangeStart: number,
   rangeEnd: number
-): [IndexIntoSamplesTable, IndexIntoSamplesTable] {
+): StartEndSampleRange {
   // TODO: This should really use bisect. samples.time is sorted.
-  const firstSample = samples.time.findIndex(t => t >= rangeStart);
-  if (firstSample === -1) {
-    return [samples.length, samples.length];
+  const sampleStart = samples.time.findIndex(t => t >= rangeStart);
+  if (sampleStart === -1) {
+    return { sampleStart: samples.length, sampleEnd: samples.length };
   }
-  const afterLastSample = samples.time
-    .slice(firstSample)
+  const sampleEnd = samples.time
+    .slice(sampleStart)
     .findIndex(t => t >= rangeEnd);
-  if (afterLastSample === -1) {
-    return [firstSample, samples.length];
+  if (sampleEnd === -1) {
+    return { sampleStart, sampleEnd: samples.length };
   }
-  return [firstSample, firstSample + afterLastSample];
+  return { sampleStart, sampleEnd: sampleEnd + sampleStart };
 }
 
+/**
+ * This function filters the samples table using the sample indexes. Use
+ * getSampleIndexRangeForSelection to get this information from the time-based
+ * range information.
+ */
 export function filterThreadSamplesToRange(
   thread: Thread,
-  rangeStart: number,
-  rangeEnd: number
+  sBegin: IndexIntoSamplesTable,
```

Comment:
Optional, slightly out of scope: It'd be nice to make these variable names consistent to `sampleStart` and `sampleEnd`. `sBegin` is probably from the original implementation of this function.

Reply:
yeah, I considered it but then thought that keeping it clear that this function doesn't change was better for review ;)
I can do it though.